gui/default/syncthing/core/syncthingController.js

```diff
@@ -1343,16 +1358,30 @@ angular.module('syncthing.core')
         $scope.directoryList = [];
 
         $scope.$watch('currentFolder.path', function (newvalue) {
-            if (newvalue && newvalue.trim().charAt(0) === '~') {
-                $scope.currentFolder.path = $scope.system.tilde + newvalue.trim().substring(1);
+            if (!(newvalue)) {
+                return;
             }
+            $scope.currentFolder.path = expandTilde(newvalue);
             $http.get(urlbase + '/system/browse', {
                 params: { current: newvalue }
             }).success(function (data) {
                 $scope.directoryList = data;
             }).error($scope.emitHTTPError);
         });
 
+        $scope.$watch('currentFolder.label', function (newvalue) {
+            if (!$scope.config.options || !$scope.config.options.defaultFolderPath || $scope.currentFolder.path) {
+                return;
+            }
+            var path = $scope.config.options.defaultFolderPath;
+            if (newvalue) {
+                path = pathJoin(path, newvalue);
+            } else {
+                path = pathJoin(path, $scope.currentFolder.id);
+            }
+            $('#folderPath').prop("value", path);
```

Comment:
I just did a mix of 1. and 4. and it throws errors at me, because `watchGroup` was introduced in AngularJS 1.3 and we are using 1.2.9. AngularJS is now at 1.6.4, is there a particular reason this was never updated?